public/src/components/helpers/buttonWithConfirmationPopup.tsx

```diff
@@ -24,66 +24,73 @@ interface ButtonWithConfirmationPopupProps extends WithStyles<typeof styles> {
   confirmationText: string,
   onConfirm: () => void,
   color?: ButtonProps["color"],
-  icon: ReactElement<SvgIconProps>
+  icon: ReactElement<SvgIconProps>,
+  disabled?: boolean
 }
 
 interface ButtonWithConfirmationPopupState {
   popoverOpen: boolean,
   anchorElement?: HTMLAnchorElement
 }
 
-class ButtonWithConfirmationPopup extends React.Component<ButtonWithConfirmationPopupProps, ButtonWithConfirmationPopupState> {state:ButtonWithConfirmationPopupState = {
-  popoverOpen: false,
-  anchorElement: undefined
-}
+class ButtonWithConfirmationPopup extends React.Component<ButtonWithConfirmationPopupProps, ButtonWithConfirmationPopupState> {
+  state: ButtonWithConfirmationPopupState = {
+    popoverOpen: false,
+    anchorElement: undefined
+  }
 
-onClick = (event: React.MouseEvent<HTMLAnchorElement>) =>  {
-  this.setState({
-    popoverOpen: true,
-    anchorElement: event.currentTarget
-  })
-}
+  onClick = (event: React.MouseEvent<HTMLAnchorElement>) =>  {
+    this.setState({
+      popoverOpen: true,
+      anchorElement: event.currentTarget
+    })
+  }
 
-handleConfirm = () => {
-  this.setState({ popoverOpen: false });
-  this.props.onConfirm();
-}
+  handleConfirm = () => {
+    this.setState({ popoverOpen: false });
+    this.props.onConfirm();
+  }
 
-handleCancel = () => {
-  this.setState({ popoverOpen: false });
-}
+  handleCancel = () => {
+    this.setState({ popoverOpen: false });
+  }
 
-render(): React.ReactNode {
-  const { classes } = this.props;
+  render(): React.ReactNode {
+    const { classes } = this.props;
 
-  return (
-    <>
-      <Button variant="contained" color={this.props.color} onClick={this.onClick} className={classes.button}>
-          {this.props.icon}&nbsp;{this.props.buttonText}
-      </Button>
-      <Popover
-        open={this.state.popoverOpen}
-        anchorEl={this.state.anchorElement}
-        anchorOrigin={{
-          vertical: 'bottom',
-          horizontal: 'center',
-        }}
-        transformOrigin={{
-          vertical: 'top',
-          horizontal: 'center',
-        }}
+    return (
+      <>
+        <Button
+          disabled={this.props.disabled}
```

Comment:
this is the actual change to this file - passing `disabled` into the `Button` component